src/Magento/FunctionalTestingFramework/Test/Util/AnnotationExtractor.php

```diff
@@ -162,6 +191,32 @@ public function validateStoryTitleUniqueness()
         }
     }
 
+    /**
+     * Validates uniqueness between Test Case ID and Titles globally
+     * @returns void
+     * @throws XmlException
+     * @return void
+     */
+    public function validateTestCaseIdTitleUniqueness()
+    {
+        $dupes = [];
+        foreach ($this->testCaseToTitleMappings as $newTitle => $files) {
+            if (count($files) > 1) {
+                $dupes[$newTitle] = "'" . implode("', '", $files) . "'";
+            }
+        }
+        if (!empty($dupes)) {
+            $message = "TestCaseId and Title pairs must be unique:\n\n";
+            foreach ($dupes as $newTitle => $tests) {
+                $testCaseTitleArray = explode(" - ", $newTitle);
+                $testCaseId = $testCaseTitleArray[0];
+                $title = $testCaseTitleArray[1];
+                $message .= "TestCaseId: '{$testCaseId}' Title: '{$title}' in Tests {$tests}\n\n";
```

Comment:
Maybe you want to tweak the error message a bit when $testCaseId is "[No TestCaseID]"? This is when there is a dupe where both don't have test case id and share the same title.

Reply:
I actually kind of like the exception message:
`TestCaseId: '[NO TESTCASEID]' Title: 'Magento Functional Testing Framework Documentation is available.' in Tests 'DevDocsTest', 'DevDocsTest2'`
The `[NO TESTCASEID]` is that gets appended to the title (meaning it propagates all the way to the allure report). I kind of like that consistency, but let me know if you feel strongly about it.

Reply:
let's fine.